### DISCUSSION TITLE
Pycharm debugging on osx

### DISCUSSION THREAD

**Discussion by rlevine** [inkstitch/inkstitch#2051] [Q&A]
I'm just setting up a dev environment for inkstitch, using the community edition of pycharm. Anyone familiar with pycharm on osx know if it's possible to debug inkstitch using pycharm's local "Attach to process..." feature, or do I need to get the pro version and do a remote debug?

I have a file named DEBUG in the project root, and I've tried various contortions, including coercing inkscape to run python from the same conda env as I'm using for inkstitch, but all my attempts to connect to the process have timed out.

Thanks!

**Comment by lexelby** [comment]
Huh, I wonder if this is some kind of firewall / network security nonsense?

I've only used the professional edition of PyCharm so I'm not sure what the difference is to the free one.

**Comment by rlevine** [comment]
Hmm. Do you use "Attach to process..." in the Run menu when you debug? Or have you set up a run/debug config to locahost?

**Reply by lexelby** [reply]
Uhhh I honestly can't remember.  I believe Ink/Stitch connects to PyCharm (or PyDev), not the other way around.

**Comment by rlevine** [comment]
More debugging. How do I debug running threads in pycharm? I want to watch it make stitches...

**Reply by lexelby** [reply]
I've never specifically dealt with trying to debug a background thread, but I thought it was just a matter of setting a breakpoint and hitting go.  I debug using the "Stitch Plan Preview" extension so that everything is one thread, rather than Params.  Does that help?

**Comment by rlevine** [comment]
Didn’t even think to try stitch plan preview. I’ve been trying to coerce
into talking to a server thread from params! Thanks!
